mreg_cli/outputmanager.py

```diff
@@ -394,14 +397,14 @@ def get_filter(self, command: str) -> tuple[str, Any, bool]:
                 try:
                     filter_re = re.compile(filter_str)
                 except re.error as exc:
-                    if "|" in command_issued:
-                        raise InputFailure(
-                            "Command parts that contain a pipe ('|') must be quoted.",
-                        ) from exc
+                    base_msg = f"Unable to compile regex '{filter_str}'"
+                    msg = build_error_message(command_issued)
+                    if msg:
+                        # Custom error messages with suggestions go on their own line
+                        raise CliError(f"{base_msg}\n{msg}") from exc
                     else:
-                        raise InputFailure(
-                            f"Unable to compile regex '{filter_str}': {exc}"
-                        ) from exc
+                        # Otherwise append to the base message
+                        raise CliError(f"{base_msg}: {exc}") from exc
```

Comment:
This is not ideal in my view. Maybe we should have some sort of `InlineErrorBuilder` that we can fall back on that prints the exception message on the same line if no other builder can be found.

Reply:
Yeah, long term that makes more sense. This is still an improvement!